src/msg.cpp

```diff
@@ -182,6 +186,33 @@ int zmq::msg_t::init_data (void *data_,
     return 0;
 }
 
+int zmq::msg_t::init_from_allocator (size_t size_, zmq::allocator_t *alloc_)
+{
+    zmq_assert (alloc_ != NULL && size_ != 0);
+
+    _u.lmsg.metadata = NULL;
+    _u.lmsg.type = type_lmsg;
```

Comment:
I think we should have a type for an allocated msg type, just like the one for zero-copy messages:
https://github.com/zeromq/libzmq/blob/master/src/msg.hpp#L197